src/DynamoCoreWpf/Views/FileTrust/FileTrustWarning.xaml.cs

```diff
@@ -162,10 +164,12 @@ internal void UpdatePopupLocation()
         internal void CleanPopup()
         {
             if (fileTrustWarningViewModel != null)
-            { 
+            {
                 fileTrustWarningViewModel.PropertyChanged -= ViewModel_PropertyChanged;
                 fileTrustWarningViewModel.DynFileDirectoryName = string.Empty;
             }
+
+            HomeWorkspaceModel.WorkspaceClosed -= CloseWarningPopup;
```

Comment:
How often would this get called? If the current workspace is closed, does the unsubscription works?

Reply:
When the file trust dialog is closed, this should be called.

Reply:
Thanks!